crates/pipeline-types/src/config.rs

```diff
@@ -204,3 +204,38 @@ pub struct ResourceConfig {
     #[serde(default)]
     pub storage_mb_max: Option<u64>,
 }
+
+/// Configuration for accessing a MySQL database service.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct ExternalServiceMysqlConfig {
+    pub hostname: String,
+    pub port: String,
+    pub user: String,
+    pub password: String,
+}
+
+/// Configuration for accessing a Postgres database service.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct ExternalServicePostgresConfig {
+    pub hostname: String,
+    pub port: String,
+    pub user: String,
+    pub password: String,
+}
+
+/// An external service's configuration.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, ToSchema)]
+pub enum ExternalServiceConfig {
```

Comment:
If the type is already called `ExternalService`, you don't need to also repeat the prefix in the inner *Config types.

Reply:
Inlined them into the Service enumeration